lib/materialist/materializer.rb

```diff
@@ -76,10 +76,19 @@ def initialize(url, klass)
           @options = klass.materialist_options
         end
 
-        def upsert
+        def upsert(retry_on_race_condition: true)
```

Comment:
What about an instance variable instead? That would avoid polluting the method arguments with an essentially private attribute.

Reply:
It might make it a tad confusing given this parameter is used in the recursion logic :/

Reply:
I see... Not a fan but ok!

Reply:
I would also agree with retrying using `retry` + instance variable rather than recursion.